src/uri/uri.cc

```diff
@@ -249,6 +249,54 @@ auto URI::path() const -> std::optional<std::string> {
   return path_;
 }
 
+auto URI::path(const std::string &path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  auto is_relative = path.starts_with("./") || path.starts_with("../");
```

Comment:
```suggestion
  auto is_relative = path.starts_with(".");
```

Isn't this enough?

Reply:
Also, this variable should be `const`

Reply:
what if I have `.github/workflows/` ?

Fun story! I started with the same approach haha